Add upload data tab (#14)
LucidtechAI/cradl-docs: get-started/upload-data-using-cli.md

## Upload documents one by one to a dataset

After the dataset is created we can start uploading [documents](../concepts/documents.md) and assign them to our dataset. Since we want to use the documents for training, we'll also provide [ground truth values](../concepts/documents.md#setting-ground-truths) that will define the correct output for the model on each document. We'll have to make sure that the field names in the ground truth match those in the field config we made for our model.

{% hint style="warning" %}

> It is important to have **correct** ground truths for each document we want to use for training. They are what will guide our model to making correct predictions. If there are mistakes in the ground truths, our model will learn to make those mistakes.

Reviewer: ```suggestion
It is important to have **correct** ground truth values for every document we use for training. Errors in the ground truth can degrade the training process as the model may learn to make the same mistakes.
```

curl -X POST 'https://api.cradl.ai/v1/documents' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJra...' \
--data-raw '{
    "content": "JVBERi0xLjQ...",
    "contentType": "application/pdf",
    "datasetId": <datasetId>,
    "groundTruth: [
      {
      "label": "total_amount",
      "value": "300.00"
      },
      {
        "label": "date",
        "value": "2020-02-28"
      }
    ]
}'
```
{% endtab %}

{% tab title="Python" %}
```python
ground_truth = [
  { 'label': 'total_amount', 'value': '300.00' },
  { 'label': 'date', 'value': '2020-02-28' }
]
document = client.create_document(b'<bytes data>', 'application/pdf', ground_truth=ground_truth, dataset_id=<datasetId>)
```
{% endtab %}
{% endtabs %}

```javascript
{
  "documentId": <documentId>,
  "contentType": "application/pdf",
  "retentionInDays": 1825,
  "createdTime": "2021-08-16T13:34:13.724393+0000",
  "updatedTime": null,
  "createdBy": <appClientId>,
  "updatedBy": null,
  "datasetId": <datasetId>,
  "groundTruth": [
    {
      "label": "total_amount",
      "value": "300.00"
    },
    {
      "label": "date",
      "value": "2020-02-28"
    }
  ]
}
```

{% hint style="success" %}

> If you upload documents without assigning them to a dataset, or if you want to assign them to a different dataset in the future, you can do so easily using the API.

Reviewer: ```suggestion
You can upload documents without assigning them to a dataset or you can assign them to a dataset later through the API..
```

Reviewer: Disclaimer: not sure if I understood this one correctly. Did you mean that you can change the dataset that a document is assigned to at a later point in time, or just that you don't have to immediately assign a document to a dataset?

Author: Both

{% endhint %}

## Upload multiple documents to a dataset

> Uploading documents one by one is useful for small tests, but for thousands of documents it can be a slow and painful experience.

Reviewer: ```suggestion
Uploading documents one by one can be useful for testing purposes, but is not recommended for large scale training datasets.
```

> To make the process smoother we recommend to use `datasets create-documents` in the [CLI](../reference/cli.md) . This allows you to upload your dataset in a fast and consistent way, without worrying about looping over all the documents yourself.

Reviewer: ```suggestion
For large-scale datasets we recommend using the `datasets create-documents` command from the [CLI](../reference/cli.md) . This allows you to upload your dataset in a fast and consistent way, without worrying about looping over all the documents yourself.
```

### Upload a folder with document and ground truth pairs

> In order to upload all the documents in a folder the following convention must be utilized:

Reviewer: ```suggestion
In order to upload all the documents in a folder, the following naming convention must be used:
```

> 1. each document in the folder must come with a ground truth as either a `.json|.yaml|.yml` file.

Reviewer: ```suggestion
1. each document in the folder must have a corresponding ground truth file in JSON or YAML format.
```

If some of your documents are missing ground-truths they will simply be skipped.
{% endhint %}

### Upload a folder with all the meta data in one file
The other alternative is to specify a file that contains all the paths and ground truths to the documents we want to upload, let us call it `upload-specification.json`.
Below is an example of how this file would look if we only want to upload two documents.
```json
{
    "path/to/document1.pdf": {
        "ground_truth": [
            {
                "label": "total_amount",
                "value": "100.00"
            },
            {
                "label": "due_date",
                "value": "2021-10-30"
            },
            {
                "label": "vendor_name",
                "value": "Company X"
            }
        ]
    },
    "path/to/document2.png": {
        "ground_truth": [
            {
                "label": "total_amount",
                "value": "200.00"
            },
            {
                "label": "due_date",
                "value": "2021-11-30"
            },
            {
                "label": "vendor_name",
                "value": "Company Y"
            }
        ]
    }
}
```

> The file is just a dictionary with the path to each document you want to upload as the keys, and their corresponding ground truth as values.

Reviewer: ```suggestion
In this file, documents and ground truth values are represented in a dictionary where each key is assumed to be a path to a document you want to upload and each value is assumed to be the corresponding ground truth value.
```

> We are now ready to upload all the documents and their ground-truth by using the `create-documents` command in the `datasets` module.

Reviewer: ```suggestion
We are now ready to upload all the documents and ground truths with the `create-documents` command.
```

```shell
las datasets create-documents <datasetId> upload-specification.json
```

> This function automatically caches your progress, so if something interrupts the call it can be called again and take off right where it ended, without having to worry about the same documents being uploaded twice.

Reviewer: ```suggestion
This command handles interruptions gracefully and can be safely restarted without having to worry about duplicate documents being uploaded.
```
